__tests__/components/HardwareConfirmationButton.spec.ts

```diff
@@ -0,0 +1,105 @@
+ *
+ */
+import HardwareConfirmationButton from '@/components/HardwareConfirmationButton/HardwareConfirmationButton.vue';
+import { HardwareConfirmationButtonTs } from '@/components/HardwareConfirmationButton/HardwareConfirmationButtonTs';
+import TrezorConnect from '@/core/utils/TrezorConnect';
+import { WalletsModel1 } from '@MOCKS/Accounts';
+import { getComponent } from '@MOCKS/Components';
+import { transferTransaction } from '@MOCKS/transactions';
+
+describe('components/HardwareConfirmationButton', () => {
+    const account = WalletsModel1;
+    const mockAccountStore = {
+        namespaced: true,
+        state: {
+            currentAccount: account,
+        },
+        getters: {
+            currentAccount: (state) => {
+                return state.currentAccount;
+            },
+        },
+    };
+
+    const getHardwareConfirmationButtonWrapper = () => {
+        return getComponent(HardwareConfirmationButton, { account: mockAccountStore }, {}, {}, {});
+    };
+
+    test('renders hardware confirmation button', () => {
+        // Arrange:
+        const wrapper = getHardwareConfirmationButtonWrapper();
+
+        // Act + Assert:
+        expect(wrapper.find('button').exists()).toBe(true);
+    });
+
+    test('success event emitted when confirm button clicked', async () => {
+        // Arrange:
+        const wrapper = getHardwareConfirmationButtonWrapper();
+
+        // Act:
+        await wrapper.find('button').trigger('click');
+
+        // Assert:
+        expect(wrapper.emitted('success')).toBeTruthy();
+    });
+
+    test('signs a transaction successfully', async () => {
+        // Arrange:
+        const wrapper = getHardwareConfirmationButtonWrapper();
+        const component = wrapper.vm as HardwareConfirmationButtonTs;
+        const transaction = transferTransaction;
+        const expectedResult = { success: true, payload: { data: 'PAYLOAD_DATA' } };
+        jest.spyOn(TrezorConnect, 'nemSignTransaction').mockResolvedValue(expectedResult);
+
+        // Act:
+        const signedTx = await component.signTransaction(transaction).toPromise();
+
+        // Assert:
+        expect(signedTx.payload).toBe(expectedResult.payload.data);
+    });
+
+    test('error on signing transaction', async () => {
+        // Arrange:
+        const wrapper = getHardwareConfirmationButtonWrapper();
+        const component = wrapper.vm as HardwareConfirmationButtonTs;
+        const transaction = transferTransaction;
+        const expectedResult = { success: false, payload: { error: 'PAYLOAD_ERROR' } };
+        jest.spyOn(TrezorConnect, 'nemSignTransaction').mockResolvedValue(expectedResult);
+
+        // Act:
+        try {
+            await component.signTransaction(transaction).toPromise();
+        } catch (error) {
+            // Assert:
+            expect(error.message).toBe(expectedResult.payload.error);
+        }
+    });
```

Comment:
this and following test will pass if no exception is thrown. 
i think chai should have construction like:
expect(() => { ... }).to.throw('message or exception');

Reply:
yea, I used it in other projects. We use jest, not chai here. I already tried something like below but it couldn't catch it. So I was looking at it and that's why I didn't add you as reviewer, since it wasn't complete yet. Thanks for reviewing though. Will address the comments 👍 

```
expect(component.signTransaction(transaction).toPromise()).rejects.toThrow(expectedResult.payload.error);
```

Reply:
Found the issue and fixed 👍